### DISCUSSION TITLE
How can I disable auto-updating of a new download of BN?

### DISCUSSION THREAD

**Discussion by gkprow** [Vector35/binaryninja-api#1868] [General] [answered]
When a fresh instance of Binary Ninja is installed and the GUI is opened, the updater attempts to reach out to the update/download server to fetch and install the most recent available stable/dev build. It would be great if there was a config option available in settings.json that would prevent that behavior from occurring.

**Comment by psifertex** [comment, marked as answer]
There's two possible ways you can achieve this. First, there's a global setting that you can disable all updates and even checking the changelog in settings.json via:

```
{
	"updates" : 
	{
		"activeContent" : false
	}
}
```

That said, this might disable more than you want, or you may wish the ability to only disable /some/ versions of BN. You can do that by separately setting the manifest for the update per-install path:

```
uppath=~/.binaryninja/update/`echo -en "/home/jordan/binaryninja"|shasum -a 256|cut -d' ' -f1`
mkdir -p $uppath
echo "{\"auto\":false,\"channel\":\"dev\",\"last_check\":`date +%s`,\"pending\":false}" > $uppath/manifest
```

If you've already got the API available you can also achieve this per-install change via:

```
python -c "import binaryninja; binaryninja.update.set_auto_updates_enabled(False)"
```
